client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -209,12 +212,18 @@ class DeltaSharingRestClient(
     }
   }
 
-  def getMetadata(table: Table): DeltaTableMetadata = {
+  def getMetadata(
+      table: Table,
+      versionAsOf: Option[Long] = None,
+      timestampAsOf: Option[String] = None): DeltaTableMetadata = {
     val encodedShareName = URLEncoder.encode(table.share, "UTF-8")
     val encodedSchemaName = URLEncoder.encode(table.schema, "UTF-8")
     val encodedTableName = URLEncoder.encode(table.name, "UTF-8")
+    val encodedParams = getEncodedMetadataParams(versionAsOf, timestampAsOf)
+
     val target = getTargetUrl(
-      s"/shares/$encodedShareName/schemas/$encodedSchemaName/tables/$encodedTableName/metadata")
+      s"/shares/$encodedShareName/schemas/$encodedSchemaName/tables/$encodedTableName/metadata" +
+        s"$encodedParams")
```

Comment:
What if this request is sent to the server that does not support these new params?
Do we need to worry about compatibility issues for this?
Consider adding a comment.